metrics/register.go

```diff
@@ -0,0 +1,71 @@
+package metrics
+
+import (
+	"fmt"
+	"sync"
+
+	"go.opencensus.io/stats/view"
+)
+
+var registeredViews = map[string][]*view.View{}
+var mutex = new(sync.Mutex)
+
+type ErrNamespace struct {
+	Namespace string
+}
+
+// ErrUnregisteredNamespace is an error for lookup of requested unregistered Namespace
+type ErrUnregisteredNamespace ErrNamespace
+
+func (e ErrUnregisteredNamespace) Error() string {
+	return fmt.Sprintf("no views found registered under Namespace %s", e.Namespace)
+}
+
+// ErrDuplicateNamespaceRegistration is an error for a Namespace that has already
+// registered views
+type ErrDuplicateNamespaceRegistration ErrNamespace
+
+func (e ErrDuplicateNamespaceRegistration) Error() string {
+	return fmt.Sprintf("duplicate registration of views by Namespace %s", e.Namespace)
+}
+
+// RegisterViews accepts a namespace and a slice of Views, which will be registered
+// with opencensus and maintained in the global registered views map
+func RegisterViews(namespace string, views ...*view.View) error {
+	mutex.Lock()
+	_, ok := registeredViews[namespace]
+	if !ok {
+		return ErrDuplicateNamespaceRegistration{Namespace: namespace}
+	}
+
+	if err := view.Register(views...); err != nil {
+		delete(registeredViews, namespace)
```

Comment:
Are we even adding anything to `registeredViews`?

Reply:
Sorry, this was an oversight from switching back from sync.Map. Corrected